### DISCUSSION TITLE
How to include data from related models/tables

### DISCUSSION THREAD

**Discussion by LemonPie22** [TappNetwork/filament-auditing#25] [Q&A]
I have a `Quote` model which has several relationships, e.g:
```
public function organisation()
{
    return $this->belongsTo(Organisation::class);
}
public function customer()
{
    return $this->belongsTo(User::class);
}
public function author()
{
    return $this->belongsTo(User::class);
}
```
My audit trail for quotes shows the column names and values in the form `author_id 1501 sales_organisation_id 1 sales_person_id 1501`, which isn't too friendly for end users.

How can I display the `name` column values from my related tables so that the output is more informative?

Ideally, I'd like to display something like `Author: Fred Jones, Sales organisation: Microsoft, Sales person: John Doe`

**Comment by andreia** [comment]
Hi @LemonPie22 

Thanks for this comment!
For now, a possible solution is to use [audit transformation](https://laravel-auditing.com/guide/audit-transformation.html) but it will save the extra data on database.

We had the same requirement a while ago. Working on a solution to allow modifying data for display. I'll let you know when it's available! 
Suggestions are welcome! :)

**Reply by andreia** [reply]
@LemonPie22 Just created a PR for it: https://github.com/TappNetwork/filament-auditing/pull/26/files
I would love to hear what you think about it! Thanks!

**Reply by LemonPie22** [reply]
Hi @andreia 

I see the latest version (v4.0.0) has the ability to [show a related column instead of foreign id](https://filamentphp.com/plugins/tapp-network-laravel-auditing#show-a-related-column-instead-of-foreign-id).

I've tried it on Filament 4 (v4.0.0-beta9) but it doesn't work for me.

<img width="601" alt="Screenshot 2025-06-25 at 13 04 36" src="https://github.com/user-attachments/assets/a71f2b18-0fba-4514-9248-73b9ae9a0941" />

This the mapping array in my `filament-auditing` config file:
```
'mapping' => [
    'sales_person_id' => [
        'model' => \App\Models\User::class,
        'field' => 'name',
        'label' => 'Salesperson',
    ],
    'quote_type_id' => [
        'model' => \App\Models\QuoteType::class,
        'field' => 'name',
        'label' => 'Quote type',
    ],
],
```
    
The audit table entry is:
```
  {
    "id": 14471,
    "user_type": "App\\Models\\User",
    "user_id": 1,
    "event": "updated",
    "auditable_type": "App\\Models\\Quote",
    "auditable_id": 104063,
    "old_values": "{\"tax_rate\":\"4.20\",\"deposit\":\"0.00\",\"discount\":\"0.00\",\"labour\":\"0.00\",\"sales_person_id\":4}",
    "new_values": "{\"tax_rate\":4.2,\"deposit\":0,\"discount\":0,\"labour\":0,\"sales_person_id\":\"5\"}",
    "url": "http://localhost:8001/livewire/update",
    "ip_address": "192.168.65.1",
    "user_agent": "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/137.0.0.0 Safari/537.36",
    "tags": null,
    "created_at": "2025-06-25 11:42:25",
    "updated_at": "2025-06-25 11:42:25"
  }
```

I've tried various other possibilities, cleared my caches, etc. but nothing makes any difference.
